setup.py

```diff
@@ -1,231 +1,261 @@
+        license='MIT',
+        ext_modules=ext_modules,
         install_requires=['numpy', 'murmurhash == 0.24', 'cymem == 1.30', 'preshed == 0.44',
-                          'thinc == 4.0.0', "text_unidecode", 'plac', 'six',
-                          'ujson', 'cloudpickle', 'sputnik == 0.5.2'],
-        setup_requires=["headers_workaround"],
-        cmdclass = {'build_ext': build_ext_subclass },
+                          'thinc == 4.0.0', 'text_unidecode', 'plac', 'six',
+                          'ujson', 'cloudpickle', 'sputnik == 0.6.2'],
+        cmdclass = {
+            'build_ext': build_ext_subclass},
     )
 
-    import headers_workaround
+    # Run build
+    cwd = os.path.abspath(os.path.dirname(__file__))
+    if not os.path.exists(os.path.join(cwd, 'PKG-INFO')):
+        # Generate Cython sources, unless building from source release
+        generate_cython()
 
-    headers_workaround.fix_venv_pypy_include()
-    headers_workaround.install_headers('murmurhash')
-    headers_workaround.install_headers('numpy')
+        # sync include dirs from native dependencies
+        include_dir = os.path.join(src_path, 'include')
+        if os.path.exists(include_dir):
+            shutil.rmtree(include_dir)
+        os.mkdir(include_dir)
 
+        import numpy
+        shutil.copytree(
+            os.path.join(numpy.get_include(), 'numpy'),
+            os.path.join(include_dir, 'numpy'))
 
-VERSION = '0.100'
-def main(modules, is_pypy):
-    language = "cpp"
-    includes = ['.', path.join(sys.prefix, 'include')]
-    if sys.platform.startswith('darwin'):
-        compile_options['other'].append('-mmacosx-version-min=10.8')
-        compile_options['other'].append('-stdlib=libc++')
-        link_options['other'].append('-lc++')
-    if use_cython:
-        cython_setup(modules, language, includes)
-    else:
-        exts = [c_ext(mn, language, includes)
-                      for mn in modules]
-        run_setup(exts)
-
-MOD_NAMES = ['spacy.parts_of_speech', 'spacy.strings',
-             'spacy.lexeme', 'spacy.vocab', 'spacy.attrs',
-             'spacy.morphology', 'spacy.tagger',
-             'spacy.syntax.stateclass', 
-             'spacy.tokenizer',
-             'spacy.syntax.parser', 
-             'spacy.syntax.transition_system',
-             'spacy.syntax.arc_eager',
-             'spacy.syntax._parse_features',
-             'spacy.gold', 'spacy.orth',
-             'spacy.tokens.doc', 'spacy.tokens.span', 'spacy.tokens.token',
-             'spacy.serialize.packer', 'spacy.serialize.huffman', 'spacy.serialize.bits',
-             'spacy.cfile', 'spacy.matcher',
-             'spacy.syntax.ner',
-             'spacy.symbols']
+        import murmurhash
+        shutil.copytree(
+            os.path.join(os.path.dirname(murmurhash.__file__), 'headers', 'murmurhash'),
+            os.path.join(include_dir, 'murmurhash'))
+
+    try:
+        setup(**metadata)
+    finally:
+        del sys.path[0]
```

Comment:
I think this removes the requirement to do `export PYTHONPATH=`pwd``, but then I was never precisely sure when that was or wasn't required.

As a point of super pedantic detail, it might be nice to implement this as a context manager?

    with mutate_path(...):
        # do stuff